templates/default/interfaces.erb

```diff
@@ -35,23 +35,23 @@ iface <%= @device %> inet <%= @type %>
 <% if @mtu -%>
   mtu <%= @mtu %>
 <% end %>
-<% if @pre_up -%>
-  pre-up <%= @pre_up %>
+<% @pre_up.each do |x| -%>
```

Comment:
You don't have to default to `[]` and remove support for `String`.

Just use here `Array(@pre_up).each` ;)